api/src/test/java/org/openmrs/api/impl/ConceptServiceImplTest.java

```diff
@@ -791,4 +791,22 @@ public void getCountOfDrugs_shouldReturnTheTotalNumberOfMatchingNumbers() {
 		assertEquals(new Integer(1),
 		    conceptService.getCountOfDrugs(phrase, conceptService.getConcept(conceptId), true, true, true));
 	}
+
+	/**
+	 * @see ConceptServiceImpl#getConceptNameTagByName(String)
+	 */
+	@Test
+	public void getConceptNameTagByName_shouldReturnNullIfNoConceptNameTagIsFound() {
+		assertNull(conceptService.getConceptNameTagByName("random-tag"));
+	}
+
+	/**
+	 * @see ConceptServiceImpl#getConceptNameTagByName(String) 
+	 */
+	@Test
+	public void getConceptNameTagByName_shouldReturnTheMatchingConceptNameTagObjectIfFound() {
+		ConceptNameTag c = conceptService.getConceptNameTag(1);
+		assertNotNull(c);
+		assertEquals(c.getId(), conceptService.getConceptNameTagByName(c.getTag()).getId());
```

Comment:
Is .getId() really needed? What if we only assert on the objects? The signature of these objects will get matched automatically (id, desc, etc) right?

Reply:
FYI its not important but makes more sense.

Reply:
No worry, I'll do that

Reply:
Done!